crates/analyzer/src/namespace/types.rs

```diff
@@ -489,6 +505,18 @@ impl Type {
             _ => None,
         }
     }
+
+    pub fn has_fixed_size(&self) -> bool {
```

Comment:
I'm not really happy with this. Ideally we would have some trait (e.g `FixedSize` or `HasFixedSize`) that `Base`, `Array` and friends would implement and then this method would look like the following pseudo code:

```
pub fn has_fixed_size(&self) -> bool {
   self.as_that_sweet_trait().has_fixed_size()
}
```
But that kind of setup does't seem to be a natural Rust pattern so even if we had an `HasFixedSize` trait we would still end up breaking the `Type` into a match arm for each type which isn't ideal (violating the open-closed principle).

So, just pointing out that I'm not happy with this API and maybe someone has a better idea how to solve that.